android/src/main/java/com/fullstory/reactnative/FullStoryModuleImpl.java

```diff
@@ -140,25 +143,42 @@ private static Map toMap(ReadableMap map) {
         return map.toHashMap();
     }
 
-    public static void createPage(String pageName, ReadableMap pageProperties) {
-        page = FS.page(pageName, toMap(pageProperties));
-    }
-
-    public static void startPage(ReadableMap pageProperties) {
-        if (page != null) {
-            page.start(toMap(pageProperties));
+    @NonNull
+    public static String getUUID() {
+		UUID uuid = UUID.randomUUID();
+        return uuid.toString();
+	}
+    
+    public static void startPage(String uuid, String pageName, ReadableMap pageProperties) {
+        UUID nonce = UUID.fromString(uuid);
+
+        try {
+            Method getWebViewClientMethod = FS.class.getMethod("__pageView", UUID.class, String.class, Map.class);
+            getWebViewClientMethod.invoke(null, nonce, pageName, toMap(pageProperties));
+        } catch (Throwable t) {
+            System.out.println("Unable to invoke __pageView via reflection, please contact FullStory Support.");
```

Comment:
This could happen if FS is disabled or if the FS version is too low. We should communicate this and probably not print the message every time. We should try to initialize method references once, the first time they are needed and cache them. We may need to detect the FS version and only attempt this if it is at least 1.38.

Reply:
For customer visible errors, I generally try to phrase errors as "broad category, impact, action" (there is no need to expose internal implementation details if there is nothing they can do about it).  So I would write this:
```suggestion
            System.out.println("Unable to access native FullStory pages API.  Pages API will not function correctly.  Make sure that your plugin is at least version 1.38; if the issue persists, please contact FullStory Support.");
```

Also, is System.out.println() the correct logging API here?

Reply:
Using android.util.Log now